driver-core/src/main/java/com/datastax/driver/core/PooledConnection.java

```diff
@@ -25,8 +25,11 @@
 
     private final HostConnectionPool pool;
 
-    /** Used in {@link HostConnectionPool} to handle races between two threads trying to trash the same connection */
-    final AtomicBoolean markForTrash = new AtomicBoolean();
+    enum State {OPEN, TRASHED, GONE }
+
+    final AtomicReference<State> state = new AtomicReference<State>(State.OPEN);
```

Comment:
It might be nice to have this exposed in some way (maybe part of the JMX work?) to see the state of connections in the pool.